### PR TITLE
Feature/general improvements

### PR DESCRIPTION
**Pull request by lrascao**
This set of commits are general quality of life minor improvements, tests were failing on MacOS and should be passing now. 
This is a WIP as it's waiting on https://github.com/rusterlium/erlang-cargo/pull/6, a second PR after that one and a Hex version cut.

### CODE REVIEW

File: rebar.config
Comment (lrascao) [inline rebar.config]: this branch is perched on top of https://github.com/rusterlium/erlang-cargo/pull/6 and will be PR'd after it

**Comment by filmor** [comment]
Ah, sorry, I didn't see the part about you waiting for a Hex release, will do that tomorrow.

**Comment by lrascao** [comment]
I guess we can wait until this one passes review

**Comment by lrascao** [comment]
i'll try and squeeze in some time this week so it can promoted out of WIP and start the review

**Comment by lrascao** [comment]
@filmor this is now ready for review, if approved, before merging we can cut a new hex version of `erlang-cargo` and fixup 95aadb3

File: src/rebar3_cargo_compile_prv.erl
Comment (filmor) [inline src/rebar3_cargo_compile_prv.erl]: No, the `crates.hrl` is not a "public" header file.

Comment (lrascao) [reply src/rebar3_cargo_compile_prv.erl]: agree, and it should be git ignored by any project using it. But if it's getting generated somewhere might as well be `include` (if possible), tipically only `.erl` files live in `src`

File: ct/rebar3_cargo_SUITE_data/test_port_app/src/complex1.erl
Comment (lrascao) [inline ct/rebar3_cargo_SUITE_data/test_port_app/src/complex1.erl]: missed this, will delete it

File: src/rebar3_cargo_compile_prv.erl
Comment (filmor) [reply src/rebar3_cargo_compile_prv.erl]: The convention seems to be to have internal headers in `src`.

Comment (lrascao) [reply src/rebar3_cargo_compile_prv.erl]: ok, i can move this commit out of this PR context, another way would be to introduce an option for the target dir of `crates.hrl` that defaults to `src`, thoughts?

**Comment by lrascao** [comment]
@filmor just checking, is something blocking this or just not enough bandwidth atm?

File: src/rebar3_cargo_compile_prv.erl
Comment (filmor) [reply src/rebar3_cargo_compile_prv.erl]: I still stand by this, please move this change out.

Comment (lrascao) [reply src/rebar3_cargo_compile_prv.erl]: 7dad8595 has been removed

Comment (lrascao) [reply src/rebar3_cargo_compile_prv.erl]: 1e489afe that contained this change has been dropped

**Comment by filmor** [comment]
Sorry for the lag, looks good, I'll go over it once more on the weekend and merge.

**Review by filmor — APPROVED** [review]

**Pull request merged**
